### PR TITLE
Add pic

### PR DESCRIPTION
**Pull request by hossam-nasr**
Functionality for adding a face pic. Implements #19 #10

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -1,22 +1,47 @@
...
 firebase.initializeApp(firebaseConfig);
 
 const db = firebase.firestore();
+const storage = firebase.storage();
 const collectionRef = db.collection("carts");
 const getQuery = db.collectionGroup("carts").orderBy("timestamp", "asc");
+const getTimestamp = firebase.firestore.Timestamp.now;
+
+const extractJsonFromResponse = async response =>
+  await Promise.all(response.docs.map(extractDocData));
```
Comment (jebking) [inline src/helpers.js]: you don't need to `return await` outside of `try/catch` blocks, right?

File: src/helpers.js [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
   getQuery
     .get()
     .then(response => {
-      const respJson = response.docs.map(
-        doc => doc && doc.exists && doc.data()
-      );
-      callback(respJson);
+      extractJsonFromResponse(response).then(respCarts => {
+        callback(respCarts);
+      });
```
Comment (jebking) [inline src/helpers.js]: you can just do `...then(callback)`

File: src/helpers.js [hunk 3]
```diff
@@ -25,20 +50,50 @@ export const getCarts = callback => {
 
 export const setupCartUpdateListener = callback => {
   getQuery.onSnapshot(response => {
-    const respCarts = response.docs.map(doc => doc && doc.exists && doc.data());
-    callback(respCarts);
+    extractJsonFromResponse(response)
+      .then(respCarts => {
```
Comment (jebking) [inline src/helpers.js]: `...then(callback)`

File: src/screens/MainScreen/components/AddCartForm/components/ImagePicker/index.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        <Modal.Header closeButton>
+          <ModalTitle>Crop your image to fit your face!</ModalTitle>
+        </Modal.Header>
+        <ModalBody>
+          <AvatarEditor
+            ref={editor}
+            image={file}
+            width={200}
+            height={267}
+            borderRadius={134}
```
Comment (jebking) [inline src/screens/MainScreen/components/AddCartForm/components/ImagePicker/index.js:110]: magic numbers

File: src/screens/MainScreen/components/AddCartForm/index.js [hunk 5]
```diff
@@ -22,6 +25,16 @@ const AddCartForm = ({ onSubmit }) => {
     setName(event.target.value);
   };
 
+  const onImageSave = blob => {
```
Comment (jebking) [inline src/screens/MainScreen/components/AddCartForm/index.js:28]: this is just a function rename

File: src/screens/MainScreen/components/Train/index.js [hunk 6]
```diff
@@ -6,8 +6,14 @@ import Cart from "./components/Cart";
 const Train = ({ carts, cartSize }) => (
   <Container>
     <HeadImg cartSize={cartSize} src={TrainHeadGif} />
-    {carts.map(({ name, color, timestamp }) => (
-      <Cart key={timestamp} name={name} color={color} cartSize={cartSize} />
+    {carts.map(({ name, color, timestamp, facepic }) => (
```
Comment (jebking) [inline src/screens/MainScreen/components/Train/index.js:9]: looks like you can make a `renderCart` function here so that it's `carts.map(renderCart)`

**Review by jebking — APPROVED** [review]
Nice, just wrote some comments for code clarity

File: src/helpers.js (hunk 1 above)
Comment (hossam-nasr) [reply src/helpers.js]: Didn't know that. Removed it

File: src/helpers.js (hunk 2 above)
Comment (hossam-nasr) [reply src/helpers.js]: Done

**Pull request merged**